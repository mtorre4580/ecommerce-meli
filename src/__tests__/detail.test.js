import React from 'react';
import Detail from '../components/detail/detail';
import renderer from 'react-test-renderer';

const responseMockDetail = {author:{name:'Matias Daniel',lastname:'Torre'},item:{id:'MLA772322281',title:'Ipod Touch 4ta Generacion 32 Gb',price:{currency:'ARS',amount:680,decimals:null},picture:'https://mla-s2-p.mlstatic.com/738532-MLA25881108346_082017-O.jpg',condition:'used',free_shipping: false,sold_quantity:0,description:'iPod Touch 4\nCapacidad 32 gB\nPantalla Retina de 3,5 pulgadas\n\nEl producto no esta funcionando\nLa pantalla se salio del chasis'},categories:['Electrónica, Audio y Video','Audio','Reproductores Portátiles','iPod','iPod touch','32 GB']};

describe('<Detail />', () => {

    it('should render snapshot', () => {
        const data = responseMockDetail;
        const loading = false;
        const error = false;
        const component = renderer.create(<Detail loading={loading} data={data} error={error} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

 });



