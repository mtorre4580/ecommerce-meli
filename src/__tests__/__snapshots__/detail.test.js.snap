// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Detail /> should render snapshot 1`] = `
<div
  className="container"
>
  <nav>
    <ol
      className="breadcrumb"
    >
      <li
        className="breadcrumb-item"
      >
        Electrónica, Audio y Video
      </li>
      <li
        className="breadcrumb-item"
      >
        Audio
      </li>
      <li
        className="breadcrumb-item"
      >
        Reproductores Portátiles
      </li>
      <li
        className="breadcrumb-item"
      >
        iPod
      </li>
      <li
        className="breadcrumb-item"
      >
        iPod touch
      </li>
      <li
        className="breadcrumb-item"
      >
        32 GB
      </li>
    </ol>
  </nav>
  <section
    className="item-detail"
  >
    <div
      className="item-detail-ctn"
    >
      <div
        className="item-detail-image"
      >
        <img
          alt="Ipod Touch 4ta Generacion 32 Gb"
          src="https://mla-s2-p.mlstatic.com/738532-MLA25881108346_082017-O.jpg"
        />
      </div>
      <div
        className="item-detail-header"
      >
        <p
          className="condition"
        >
          <span>
            Usado
             - 
          </span>
          <span>
            0
             vendidos
          </span>
        </p>
        <h2
          className="item-detail-title"
        >
          Ipod Touch 4ta Generacion 32 Gb
        </h2>
        <p
          className="price"
        >
          <span>
            $
          </span>
          680
        </p>
        <button
          className="btn btn-item-buy"
          type="button"
        >
          Comprar
        </button>
      </div>
    </div>
    <div
      className="item-description-ctn"
    >
      <h3>
        Descripción del producto
      </h3>
      <div
        className="item-description"
      >
        iPod Touch 4
Capacidad 32 gB
Pantalla Retina de 3,5 pulgadas

El producto no esta funcionando
La pantalla se salio del chasis
      </div>
    </div>
  </section>
</div>
`;
